test/integration/hds_integration_test.cc

```diff
@@ -125,30 +159,61 @@ class HdsIntegrationTest : public Grpc::VersionedGrpcClientIntegrationParamTest,
+      bool use_tls = false) {
     envoy::service::health::v3::HealthCheckSpecifier server_health_check_specifier_;
     server_health_check_specifier_.mutable_interval()->set_nanos(100000000); // 0.1 seconds
 
-    auto* health_check = server_health_check_specifier_.add_cluster_health_checks();
+    auto* cluster_health_check = server_health_check_specifier_.add_cluster_health_checks();
 
-    health_check->set_cluster_name("anna");
+    cluster_health_check->set_cluster_name("anna");
     Network::Utility::addressToProtobufAddress(
         *host_upstream_->localAddress(),
-        *health_check->add_locality_endpoints()->add_endpoints()->mutable_address());
-    health_check->mutable_locality_endpoints(0)->mutable_locality()->set_region("middle_earth");
-    health_check->mutable_locality_endpoints(0)->mutable_locality()->set_zone("shire");
-    health_check->mutable_locality_endpoints(0)->mutable_locality()->set_sub_zone("hobbiton");
-
-    health_check->add_health_checks()->mutable_timeout()->set_seconds(MaxTimeout);
-    health_check->mutable_health_checks(0)->mutable_interval()->set_seconds(MaxTimeout);
-    health_check->mutable_health_checks(0)->mutable_unhealthy_threshold()->set_value(2);
-    health_check->mutable_health_checks(0)->mutable_healthy_threshold()->set_value(2);
-    health_check->mutable_health_checks(0)->mutable_grpc_health_check();
-    health_check->mutable_health_checks(0)
-        ->mutable_http_health_check()
-        ->set_hidden_envoy_deprecated_use_http2(false);
-    health_check->mutable_health_checks(0)->mutable_http_health_check()->set_path("/healthcheck");
-
+        *cluster_health_check->add_locality_endpoints()->add_endpoints()->mutable_address());
+    cluster_health_check->mutable_locality_endpoints(0)->mutable_locality()->set_region(
+        "middle_earth");
+    cluster_health_check->mutable_locality_endpoints(0)->mutable_locality()->set_zone("shire");
+    cluster_health_check->mutable_locality_endpoints(0)->mutable_locality()->set_sub_zone(
+        "hobbiton");
+    auto* health_check = cluster_health_check->add_health_checks();
+    health_check->mutable_timeout()->set_seconds(MaxTimeout);
+    health_check->mutable_interval()->set_seconds(MaxTimeout);
+    health_check->mutable_unhealthy_threshold()->set_value(2);
+    health_check->mutable_healthy_threshold()->set_value(2);
+    health_check->mutable_grpc_health_check();
+    auto* http_health_check = health_check->mutable_http_health_check();
+    http_health_check->set_path("/healthcheck");
+    http_health_check->set_codec_client_type(codec_type);
+    if (use_tls) {
+      // set bool for use in struct.
+      ProtobufWkt::Value v;
+      v.set_string_value("true");
+
+      // map our transport socket matches with our matcher.
+      auto* match_filter = health_check->mutable_transport_socket_match_criteria();
+      match_filter->mutable_fields()->insert({"good_match", v});
+
+      // create the list of all possible matches.
```

Comment:
Same comment as above on making more use of YAML rather than having to manually build the transport socket match criteria programatically.

Reply:
I did two separate merges of yaml as explained above in hds_test here as well.